4788/src/main/include/MagLoader2.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "strategy/StrategySystem.h"
+#include "RobotMap.h"
+
+enum class MagLoaderState {
+  IDLE,
+  MANUAL,
+  SHIFTUP,
+  SHIFTDOWN,
+  FIRE,
+  AUTO
+};
```

Comment:
These can be refined. Think about the specific states that the magazine will be in:
- Manual control
- Auto control
  - Waiting for balls
  - Shift a ball in
- Fire balls

In this case, your states might look like the following:
```suggestion
enum class MagLoaderState {
  AUTO,
  SHIFT_IN,
  MANUAL,
  FIRE
};
```

In this case, the default state is `AUTO`. In `AUTO` mode, it sits idle until a ball is detected, at which point it switches state to `SHIFT_IN`. When `SHIFT_IN` has finished shifting the ball, it then goes back to `AUTO` to wait for another ball.

```
AUTO --> SHIFT_IN    (ball detected)
SHIFT_IN --> AUTO    (shift complete)
```

This provides an automatic transition that allows the system to operate according to what the sensors detect, without having to do all that logic in the strategy